File: tools/dev-dashboard/check-models.js
// tools/dev-dashboard/check-models.js
const path = require('path');
require('dotenv').config({ path: path.join(__dirname, '.env') });
const { GoogleGenerativeAI } = require("@google/generative-ai");

async function check() {
    console.log("🔍 APIキー:", process.env.GEMINI_API_KEY ? "読み込みOK" : "❌ 未設定");
    
    const genAI = new GoogleGenerativeAI(process.env.GEMINI_API_KEY);
    
    try {
        console.log("📡 利用可能なモデルを取得中...");
        // 利用可能なモデル一覧を取得する
        const modelResponse = await genAI.getGenerativeModel({ model: "gemini-1.5-flash" }); 
        // ※ 本来は listModels() を使いたいが、SDKのバージョンによっては複雑なため
        //   まずは一番推奨されるモデルで疎通確認だけ行います。
        
        const result = await modelResponse.generateContent("Hello");
        console.log("✅ 接続成功！ gemini-1.5-flash は使用可能です。");
        console.log("🤖 返答:", result.response.text());
        
    } catch (error) {
        console.error("❌ エラー詳細:", error.message);
        console.log("\n💡 ヒント: APIキーが正しいか、Google AI Studioでキーが有効になっているか確認してください。");
    }
}

check();

File: memory_bank_scripts/check_models.py
import google.generativeai as genai
import os
from dotenv import load_dotenv

# 環境変数を読み込む
load_dotenv()

# APIキーを設定
api_key = os.getenv("GOOGLE_API_KEY")
if not api_key:
    print("Error: GOOGLE_API_KEYが見つかりません。.envを確認してください。")
else:
    genai.configure(api_key=api_key)

    print("=== あなたの環境で利用可能なGeminiモデル一覧 ===")
    try:
        # モデル一覧を取得して表示
        for m in genai.list_models():
            # "generateContent"（会話・文章生成）に対応しているモデルだけ表示
            if 'generateContent' in m.supported_generation_methods:
                print(f"- {m.name}")
    except Exception as e:
        print(f"エラーが発生しました: {e}")